### PR TITLE
[Improvement]: Move grid data related functions to admin-ui-classic-bundle

### LINKED ISSUE #15971: Move function to admin-ui-classic-bundle
### Improvement description

https://github.com/pimcore/pimcore/blob/0fcf252faafdaf270d620d1cc96bb49fc14dabcc/models/Asset/Service.php#L277

doesn't belong to pimcore/pimcore it should be in https://github.com/pimcore/admin-ui-classic-bundle

### PR DESCRIPTION
**Pull request by kingjia90**
## Changes in this pull request  
Resolves https://github.com/pimcore/pimcore/issues/15971
Requires https://github.com/pimcore/admin-ui-classic-bundle/pull/457

### CODE REVIEW

**Comment by kingjia90** [comment]
The combination pimcore `^11.2` and admin ui `^1.4` seem it would work fine (test green) as https://github.com/pimcore/admin-ui-classic-bundle/pull/457/commits/27f998c8e094bf79e9fd8552111c73967d6058c2
Just a little tricky to test on pimcore/pimcore as there are no hard requirement for which version of admin ui it should use, outside the platform version

File: models/DataObject/Service.php [hunk 1]
```diff
@@ -290,191 +291,12 @@ public static function isHelperGridColumnConfig(string $field): bool
     }
 
     /**
-     * Language only user for classification store !!!
-     *
-     *
-     *
+     * @deprecated Despite being internal, keeping in here to avoid bundles to require 11.2 or set a compatibility layer when bundles support both 10 & 11
```
Comment (kingjia90) [inline models/DataObject/Service.php]: After a second scan, there are no bundles that support both 10 and 11 that would be affected, so either keep it as is, or bump the minimum requirement to `11.2` in the following bundles, or compatibility layer with class_exist on bundle side
- [ ] https://github.com/pimcore/customer-data-framework/blob/1b0ee60083618f9a2d446dc618f5f256af8e2c7b/src/GDPR/DataProvider/Customers.php#L201
- [ ] https://github.com/pimcore/advanced-object-search/blob/b67be0a7e34b2470ac21e5fe73c5dcb85c1ad2fd/src/Controller/AdminController.php#L176

**Comment by blankse** [comment]
@kingjia90 So it is not possible to use pimcore 11.2 with admin ui < 1.4? So I think you should add it to the conflict section in the composer.json

**Comment by kingjia90** [comment]
@blankse that's a nice idea, thank you!

**Comment by blankse** [comment]
Maybe we should also add the admin ui to the require-dev section to fix the phpstan tests? There are also many errors in the baseline because the bundle is missing.

File: models/DataObject/Service.php [hunk 2]
```diff
@@ -552,20 +374,6 @@ public static function calculateCellValue(AbstractObject $object, array $helperD
         });
     }
 
-    public static function getHelperDefinitions(): array
```
Comment (NiklasBr) [inline models/DataObject/Service.php:556]: BC break here.

Comment (kingjia90) [reply models/DataObject/Service.php:556]: Thank you! 

Resolved by https://github.com/pimcore/pimcore/pull/16745/commits/351fb2a09a5bb589b2ef016705aaefbaaa1578ae and the upgrade note

**Comment by markus-moser** [comment]
@kingjia90 Could you please resolve the conflicts?

**Comment by kingjia90** [comment]
> @kingjia90 Could you please resolve the conflicts?

Done, more specifically the changes from https://github.com/pimcore/pimcore/pull/16806 and https://github.com/pimcore/pimcore/pull/17006/ weren't merged there when i cut/pasted across repos at the time, but now are moved to https://github.com/pimcore/admin-ui-classic-bundle/pull/457 accordingly

File: models/DataObject/Service.php (hunk 1 above)
Comment (markus-moser) [reply models/DataObject/Service.php]: Let's keep it as it is for the moment 👍

File: composer.json [hunk 3]
```diff
@@ -127,12 +127,14 @@
     "twig/extra-bundle": "^3.4.0",
     "twig/string-extra": "^3.3.5",
     "twig/twig": "^3.3.9",
-    "umpirsky/country-list": "^2.0.6"
+    "umpirsky/country-list": "^2.0.6",
+    "pimcore/admin-ui-classic-bundle": "dev-move-previewthumbfromcore as 1.4"
```
Comment (markus-moser) [inline composer.json]: Can we remove this?

**Review by markus-moser — APPROVED** [review]
LGTM, we need to check the composer.json (see my comment)

File: composer.json [hunk 4]
```diff
@@ -133,6 +133,7 @@
     "symfony/symfony": "*",
     "sabre/dav": "4.2.2",
     "thecodingmachine/safe": "<2.0",
+    "pimcore/admin-ui-classic-bundle": "<1.4",
```
Comment (blankse) [inline composer.json]: Should be <1.5 now
```suggestion
    "pimcore/admin-ui-classic-bundle": "<1.5",
```

**Comment by kingjia90** [comment]
So current state is all the tests are green here and https://github.com/pimcore/admin-ui-classic-bundle/pull/457/commits/27f998c8e094bf79e9fd8552111c73967d6058c2

I was trying to make the core less depending on what version of admin ui bundle is installed.
Despite the internal stuff that could be deleted, they are deprecated.

The downsides are that there are `class_exists` checks and we are keeping `DataObject\Service::gridObjectData`(it's used in ManyToManyObject class definition somehow) and disabling the search tests for documents and assets, if one is not on admin-ui `1.5+`, would it be a good compromise?

If yes, the upgrade notes needs to be improved, if no, i can reset the branch to the previously approved commit

**Comment by markus-moser** [comment]
@kingjia90 Ok, let's solve it like this 👍

**Pull request merged**